### DISCUSSION TITLE
Controlling line thickness by vertices possible?

### DISCUSSION THREAD

**Discussion by vidur501** [bnpr/Malt#444] [Q&A] [answered]
Is it possible to control the line thickness by mesh geometry, e.g. having maximum line thickness at the vertices of an object and letting the lines get thinner the farther they are away from the vertices?

Something like this:
![lines_example](https://user-images.githubusercontent.com/100998713/201443413-61f3a67e-35cd-4e3d-b452-fee4c124b20a.jpg)

**Comment by Kolupsy** [comment]
Right now I can not think of a way of achieving this procedurally but there would be ways and methods that are a little more involved to achieve something like this.

**Comment by pragma37** [comment, marked as answer]
There's no way to retrieve the distance to the vertex from a shader.
This is not something specific to Malt, though.

It can be worked around using geometry shaders, but they're currently not supported in Malt.

For reproducing the effect from your image I think your best option is to store a "line width scale" in a vertex color.
You can subdivide the mesh in "simple" mode and set the "original" vertices to white and the new subdivided vertices to black, then you can use that for scaling the line width.
I guess it should be possible to do it automatically using geometry nodes?


![imagen](https://user-images.githubusercontent.com/36610999/201482359-c80e5b54-d077-4a56-9133-f841df4dd3b6.png)
[vertex-line-scale.zip](https://github.com/bnpr/Malt/files/9995315/vertex-line-scale.zip)
